### PR TITLE
Upgrade almost all packages to latest available version

### PR DESCRIPTION
**Pull request by fjtrujy**
# Description

This PR just basically tries to update all the packages to the latest available version. Most of them are now being compiled using `cmake` without any additional change.

Other packages have been added as well.

A script for installing all the available packages in the latest release has been created for helping in the process of the toolchain installation.

Finally, the CI/CD system has been updated for using the `pspsdk` docker image and also generating the next docker layer.

## POSSIBLE BREAKING CHANGES
How most of the libraries have been updated to the latest version, they could include some major changes that would require some updates in the code.

### MORE INFO
This PR is part of a collection of PRs, where the whole `pspdev` environment for development is being improved, with the main objectives of getting a more POSIX & Standard toolchain and integrating a CI/CD workflow to increase testability and reliability.

Others PRs related:
https://github.com/pspdev/psp-pacman/pull/15
https://github.com/pspdev/psptoolchain-extra/pull/1
https://github.com/pspdev/psptoolchain-allegrex/pull/1
https://github.com/pspdev/psptoolchain/pull/160
https://github.com/pspdev/pspsdk/pull/58

Here you have more info about the new repo structures, artifacts, and docker layers generated:

layers

Thanks

### CODE REVIEW

File: argtable2/PSPBUILD
Comment (sharkwouter) [inline argtable2/PSPBUILD:7]: License cannot be empty

File: argtable2/PSPBUILD
Comment (sharkwouter) [inline argtable2/PSPBUILD:4]: Remove ``.`` at the end of the sentence

File: bzip2/PSPBUILD
Comment (sharkwouter) [inline bzip2/PSPBUILD:12]: The upstream tar.gz for this version and a patch file would be preferred here.

File: enet/PSPBUILD
Comment (sharkwouter) [inline enet/PSPBUILD:7]: license cannot be empty

File: enet/PSPBUILD
Comment (sharkwouter) [inline enet/PSPBUILD:11]: This commit is not version 1.3.17. Please use the tar.gz for the tag or make the version include the commit hash.

File: expat/PSPBUILD
Comment (sharkwouter) [inline expat/PSPBUILD:12]: Please use the tar.gz for this version if available.

File: freetype2/PSPBUILD
Comment (sharkwouter) [inline freetype2/PSPBUILD:13]: Please use the tar.gz for this version if available.

File: jpeg/PSPBUILD
Comment (sharkwouter) [inline jpeg/PSPBUILD:3]: This change requires ``epoch=1`` to be added to the PSPBUILD, otherwise psp-pacman cannot upgrade from the previous version.

File: jpeg/PSPBUILD
Comment (sharkwouter) [inline jpeg/PSPBUILD:12]: Please use the tar.gz for this version if available.

File: freetype2/PSPBUILD
Comment (sharkwouter) [inline freetype2/PSPBUILD:1]: Please remove

File: libconfuse/PSPBUILD
Comment (sharkwouter) [inline libconfuse/PSPBUILD:1]: Please remove

File: libconfuse/PSPBUILD
Comment (sharkwouter) [inline libconfuse/PSPBUILD:5]: Remove ``.`` at the end

File: libconfuse/PSPBUILD
Comment (sharkwouter) [inline libconfuse/PSPBUILD:12]: Preferably use a patch file and the official tar.gz

File: libmikmod/PSPBUILD
Comment (sharkwouter) [inline libmikmod/PSPBUILD:12]: Please use a tar.gz or add the commit to the version number

File: libogg/PSPBUILD
Comment (sharkwouter) [inline libogg/PSPBUILD:12]: Please use the tar.gz or add the commit to the version number

File: libogg/PSPBUILD
Comment (sharkwouter) [inline libogg/PSPBUILD:1]: Please remove

File: libpng/PSPBUILD
Comment (sharkwouter) [inline libpng/PSPBUILD:12]: Please use the official versioned tar.gz or add the commit to the version number

File: libvorbis/PSPBUILD
Comment (sharkwouter) [inline libvorbis/PSPBUILD:1]: Please remove

File: libvorbis/PSPBUILD
Comment (sharkwouter) [inline libvorbis/PSPBUILD:12]: Please use the tar.gz or add the commit hash to the version number

File: libxmp/PSPBUILD
Comment (sharkwouter) [inline libxmp/PSPBUILD:7]: Missing license

File: libxmp/PSPBUILD
Comment (sharkwouter) [inline libxmp/PSPBUILD:11]: Please use the upstream versioned release tar.gz

File: libyaml/PSPBUILD
Comment (sharkwouter) [inline libyaml/PSPBUILD:1]: Please remove

File: libyaml/PSPBUILD
Comment (sharkwouter) [inline libyaml/PSPBUILD:12]: tar.gz would be prefered

File: opentri/PSPBUILD
Comment (sharkwouter) [inline opentri/PSPBUILD:1]: Please remove

File: opentri/PSPBUILD
Comment (sharkwouter) [inline opentri/PSPBUILD:12]: Might be worth adding to pspdev

File: opentri/PSPBUILD
Comment (sharkwouter) [inline opentri/PSPBUILD:15]: This is not optional unless the commit is set in source, which is preferable over using pkgver anyway, since we can't know if future versions will build with the same PSPBUILD and if they will work. If pkgver is used, we cannot guarantee that the build package will work.

File: pixman/PSPBUILD
Comment (sharkwouter) [inline pixman/PSPBUILD:12]: Please use the upstream tar.gz

File: pixman/PSPBUILD
Comment (sharkwouter) [inline pixman/PSPBUILD:1]: Please remove

File: pthread-embedded/PSPBUILD
Comment (sharkwouter) [inline pthread-embedded/PSPBUILD:1]: Remove empty line

File: pthread-embedded/PSPBUILD
Comment (sharkwouter) [inline pthread-embedded/PSPBUILD:2]: Please remove

File: sdl2/PSPBUILD
Comment (sharkwouter) [inline sdl2/PSPBUILD:13]: This still needs a PR for the SDL repo, but it is not a blocker for this PR.

File: smpeg/PSPBUILD
Comment (sharkwouter) [inline smpeg/PSPBUILD:35]: Nice find, damn

File: sqlite/PSPBUILD
Comment (sharkwouter) [inline sqlite/PSPBUILD:29]: The pkgrel wasn't updated

File: smpeg/PSPBUILD
Comment (sharkwouter) [inline smpeg/PSPBUILD:33]: Do update the pkgrel when changes are made

File: sdl2/PSPBUILD
Comment (sharkwouter) [inline sdl2/PSPBUILD:13]: Do update pkgrel if changes are made without the version changing

File: sdl2-mixer/PSPBUILD
Comment (sharkwouter) [inline sdl2-mixer/PSPBUILD:9]: pkgrel update required

File: sdl-ttf/PSPBUILD
Comment (sharkwouter) [inline sdl-ttf/PSPBUILD:18]: pkgrel update required

File: sdl-mixer/PSPBUILD
Comment (sharkwouter) [inline sdl-mixer/PSPBUILD:9]: pkgrel update required

File: angelscript/PSPBUILD
Comment (sharkwouter) [inline angelscript/PSPBUILD:10]: The pkgrel for the package should be increased

File: expat/PSPBUILD
Comment (sharkwouter) [inline expat/PSPBUILD:9]: The pkgrel for this package should probably be increased, depending on if the version number is changed or not.

File: libmikmod/PSPBUILD
Comment (sharkwouter) [inline libmikmod/PSPBUILD:11]: Please increase the pkgrel if the package version isn't changed

File: squirrel/PSPBUILD
Comment (sharkwouter) [inline squirrel/PSPBUILD:12]: Please use the upstream tar.gz if possible

File: zlib/PSPBUILD
Comment (sharkwouter) [inline zlib/PSPBUILD:12]: Please use the upstream tar.gz if possible

File: zziplib/PSPBUILD
Comment (sharkwouter) [inline zziplib/PSPBUILD:12]: Please use the upstream tar.gz if possible

**Review by sharkwouter — CHANGES_REQUESTED** [review]
Thanks a lot for all this work, this is a lot and it's a big deal. In addition to that it's a big improvement. The changes I request are mostly nitpicks, but I think they are important. We should make sure all packages in this repo meet the requirements set in ``CONTRIBUTING.md``. On that topic, could you in addition to the requested changes remove this line from ``CONTRIBUTING.md``?: https://github.com/pspdev/psp-packages/blob/79054a6e260d55ec8aaad92ea8e254fb586f1870/CONTRIBUTING.md?plain=1#L18

File: bzip2/PSPBUILD
Comment (fjtrujy) [reply bzip2/PSPBUILD:12]: I have created a PR to the original repo, it is waiting to be merged. Let's keep it like this, and I will update it once, it is merged

File: argtable2/PSPBUILD
Comment (fjtrujy) [reply argtable2/PSPBUILD:7]: Done

File: argtable2/PSPBUILD
Comment (fjtrujy) [reply argtable2/PSPBUILD:4]: Done

File: enet/PSPBUILD
Comment (fjtrujy) [reply enet/PSPBUILD:7]: Done

File: enet/PSPBUILD
Comment (fjtrujy) [reply enet/PSPBUILD]: Done

File: expat/PSPBUILD
Comment (fjtrujy) [reply expat/PSPBUILD:9]: Done

File: libmikmod/PSPBUILD
Comment (fjtrujy) [reply libmikmod/PSPBUILD:11]: Done

File: squirrel/PSPBUILD
Comment (fjtrujy) [reply squirrel/PSPBUILD:12]: It is not possible, the latest release was done in 2016, and more than 100 commits has been done, needed to be PSP able to compile

File: zlib/PSPBUILD
Comment (fjtrujy) [reply zlib/PSPBUILD]: done

File: zziplib/PSPBUILD
Comment (fjtrujy) [reply zziplib/PSPBUILD:12]: I can't because the needed changes to make PSP to compile is there, I have done a PR for it, and they merged already.
So until they do a new version we can't use the `tar.gz`

**Comment by fjtrujy** [comment]
Thanks for your suggestion @sharkwouter, I have already applied all of them

File: .github/workflows/build.yml
Comment (sharkwouter) [inline .github/workflows/build.yml:27]: This will still require a change before merging

File: .github/workflows/build.yml
Comment (sharkwouter) [inline .github/workflows/build.yml:58]: Change to pspdev

File: .github/workflows/build.yml
Comment (sharkwouter) [inline .github/workflows/build.yml:100]: Don't forget this one either

File: bzip2/PSPBUILD
Comment (sharkwouter) [reply bzip2/PSPBUILD:12]: That makes sense, thanks for the info

File: zziplib/PSPBUILD
Comment (sharkwouter) [reply zziplib/PSPBUILD:12]: Makes sense, thanks

**Review by sharkwouter — CHANGES_REQUESTED** [review]
I just added a small reminder on places where the container used should be changed before merging.

**Comment by sharkwouter** [comment]
> Thanks for your suggestion @sharkwouter, I have already applied all of them

There are still a couple open, not sure why Github hides them :/

**Comment by fjtrujy** [comment]
> > Thanks for your suggestion @sharkwouter, I have already applied all of them
> 
> 
> 
> There are still a couple open, not sure why Github hides them :/

You need to click in "resolve conversation", I didn't click in these ones because I didn't/couldn't do what you asked for.

File: freetype2/PSPBUILD
Comment (carstene1ns) [inline freetype2/PSPBUILD:32]: typo.

File: libmikmod/PSPBUILD
Comment (carstene1ns) [inline libmikmod/PSPBUILD:19]: Does this even build the psp version?

File: jpeg/PSPBUILD
Comment (sharkwouter) [reply jpeg/PSPBUILD:2]: Using epoch is not possible if we want to build with the github ci system. It does not support uploading files with ``:`` in the name as artifacts. Makes some sense, since Windows does not support that.

**Review by sharkwouter — APPROVED** [review]
With the current changes, the PSPBUILDs are in a good enough state to merge. Just the containers used in the workflow need a change, but this won't be merged until after pspsdk has been merged.

**Comment by fjtrujy** [comment]
> With the current changes, the PSPBUILDs are in a good enough state to merge. Just the containers used in the workflow need a change, but this won't be merged until after pspsdk has been merged.

Hello @sharkwouter ,
I have updated already the docker references.

Could you merge it?

**Comment by sharkwouter** [comment]
Thanks! Will merge when the CI has passed

**Pull request merged**
